build/patches/Add-lifetime-options-permissions.patch

```diff
@@ -0,0 +1,1415 @@
+   for (auto it = range.first; it != range.second; ++it)
+-    it->second->PermissionDenied();
++    it->second->PermissionDenied(is_one_time, lifetime_option);
+ }
+ 
+ void PermissionRequestManager::CancelledIncludingDuplicates(
+@@ -1044,7 +1056,7 @@ void PermissionRequestManager::LogWarningToConsole(const char* message) {
+ void PermissionRequestManager::DoAutoResponseForTesting() {
+   switch (auto_response_for_test_) {
+     case ACCEPT_ONCE:
+-      AcceptThisTime();
++      AcceptThisTime(content_settings::LifetimeMode::Always);
+       break;
+     case ACCEPT_ALL:
+       Accept();
+diff --git a/components/permissions/permission_request_manager.h b/components/permissions/permission_request_manager.h
+--- a/components/permissions/permission_request_manager.h
++++ b/components/permissions/permission_request_manager.h
+@@ -136,8 +136,10 @@ class PermissionRequestManager
+   GURL GetRequestingOrigin() const override;
+   GURL GetEmbeddingOrigin() const override;
+   void Accept() override;
+-  void AcceptThisTime() override;
++  void AcceptThisTime(content_settings::LifetimeMode lifetime_option) override;
+   void Deny() override;
++  void Deny_(bool is_one_time, content_settings::LifetimeMode lifetime_option);
++  void DenyThisTime(content_settings::LifetimeMode lifetime_option) override;
+   void Dismiss() override;
+   void Ignore() override;
+   bool WasCurrentRequestAlreadyDisplayed() override;
+@@ -268,9 +270,12 @@ class PermissionRequestManager
+ 
+   // Calls PermissionGranted on a request and all its duplicates.
+   void PermissionGrantedIncludingDuplicates(PermissionRequest* request,
+-                                            bool is_one_time);
++                                            bool is_one_time,
++                                            content_settings::LifetimeMode lifetime_option);
+   // Calls PermissionDenied on a request and all its duplicates.
+-  void PermissionDeniedIncludingDuplicates(PermissionRequest* request);
++  void PermissionDeniedIncludingDuplicates(PermissionRequest* request,
++                                           bool is_one_time,
++                                           content_settings::LifetimeMode lifetime_option);
+   // Calls Cancelled on a request and all its duplicates.
+   void CancelledIncludingDuplicates(PermissionRequest* request);
+   // Calls RequestFinished on a request and all its duplicates.
+diff --git a/media/base/media_switches.cc b/media/base/media_switches.cc
+--- a/media/base/media_switches.cc
++++ b/media/base/media_switches.cc
+@@ -469,7 +469,7 @@ const base::Feature kUseR16Texture{"use-r16-texture",
+ // Enables the Unified Autoplay policy by overriding the platform's default
+ // autoplay policy.
+ const base::Feature kUnifiedAutoplay{"UnifiedAutoplay",
+-                                     base::FEATURE_DISABLED_BY_DEFAULT};
++                                     base::FEATURE_ENABLED_BY_DEFAULT};
```

Comment:
Rather than this change here, the patch that disables `kUnifiedAutoplay` should be removed.

Reply:
sure, it is that in my repo I tend not to modify the official patches anymore so as not to mess up. 
if you accept it we'll fix it.

Reply:
I would like to know if there are any remaining downsides in `kUInifiedAutoplay` being enabled (see question in main discussion).